src/pybind/rados/setup.py

```diff
@@ -8,43 +12,48 @@
 import tempfile
 import textwrap
 
+
+def filter_unsupported_flags(compiler, flags):
+    if 'clang' in compiler:
+        return filterfalse(lambda f:
+                           f in ('-mcet',
+                                 '-fstack-clash-protection',
+                                 '-fno-var-tracking-assignments',
+                                 '-Wno-deprecated-register',
+                                 '-Wno-gnu-designator') or
+                           f.startswith('-fcf-protection'),
+                           flags)
+    else:
+        return flags
+
+
+def monkey_with_compiler(customize):
+    def patched(compiler):
+        customize(compiler)
+        if compiler.compiler_type != 'unix':
+            return
+        compiler.compiler[1:] = \
+            filter_unsupported_flags(compiler.compiler[0],
+                                     compiler.compiler[1:])
+        compiler.compiler_so[1:] = \
+            filter_unsupported_flags(compiler.compiler_so[0],
+                                     compiler.compiler_so[1:])
+    return patched
+
+
+distutils.sysconfig.customize_compiler = \
+    monkey_with_compiler(distutils.sysconfig.customize_compiler)
+
 if not pkgutil.find_loader('setuptools'):
```

Comment:
If possible/convenient it might be good to pull it out into its own file, but since they're separate modules I'm not sure if they should share code.

Reply:
@adamemerson i've being poking around trying to extract the the duplicated code into a module, but it turns out it's difficult to do so. because `setup.py` is a script, and it does not belong to a package. in this case, it's not possible to use relative import. and i don't want to change `sys.path` or mess around with `__package__`. so probably we need to live with this unless any of us has a solution.

Reply:
Makes sense. That's basically why I didn't do it, though I also don't know Python all /that/ well so I thought I might have missed a trick.